components/micro-gateway-core/src/main/ballerina/src/gateway/listeners/api_gateway_listener.bal

```diff
@@ -123,11 +123,19 @@ public function getAuthHandlers() returns http:InboundAuthHandler[] {
         },
         verifyHostname: getConfigBooleanValue(HTTP_CLIENTS_INSTANCE_ID, ENABLE_HOSTNAME_VERIFICATION, true)
     };
+
     http:OutboundAuthConfig? auth = ();
+    // support backward compatibility in reading the basic auth credentials when connecting with KM.
+    string username = getConfigValue(KM_CONF_INSTANCE_ID, USERNAME, "");
+    string password = getConfigValue(KM_CONF_INSTANCE_ID, PASSWORD, "");
+    if (username.length() == 0 && password.length() == 0) {
+        username = getConfigValue(KM_CONF_SECURITY_BASIC_INSTANCE_ID, USERNAME, "admin");
```

Comment:
With amali's PR #948 for default configs, we have constants for default values of username and password. Let's use those after merging that PR.